### PR TITLE
feat(iHeartRadio): add site support update presence

### PR DESCRIPTION
**Pull request by iiTxicz**
## Description 
Added new Site Support (https://listen.iheart.com/)
Remodeled entire architecture of presence
#8150 - Giving more support towards this request

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>
https://imgur.com/a/hp6A0rr 

</details>

### CODE REVIEW

**Comment by iiTxicz** [comment]
Also requesting I take over the Presence due to entire makeover over the presence.

**Comment by iiTxicz** [comment]
> Also requesting I take over the Presence due to entire makeover over the presence.

Let me know if I can change it to my name before concluding the PR.

File: websites/I/iHeartRadio/metadata.json [hunk 1]
```diff
@@ -1,29 +1,27 @@
 {
 	"$schema": "https://schemas.premid.app/metadata/1.10",
 	"author": {
-		"name": "Striker",
```
Comment (darkvillager2) [inline websites/I/iHeartRadio/metadata.json:4]: Do add him as contributor.

Comment (iiTxicz) [reply websites/I/iHeartRadio/metadata.json:4]: Done

File: websites/I/iHeartRadio/presence.ts [hunk 2]
```diff
@@ -1,194 +1,757 @@
...
-	author,
-	song,
-	subtitle;
+let stationDescription: string | null = null;
 
 presence.on("UpdateData", async () => {
 	const presenceData: PresenceData = {
-		largeImageKey:
-			"https://cdn.rcd.gg/PreMiD/websites/I/iHeartRadio/assets/logo.png",
+		largeImageKey: "https://i.imgur.com/Z5CAqIw.png",
```
Comment (Bas950) [inline websites/I/iHeartRadio/presence.ts:34]:
```suggestion
		largeImageKey: baseImage,
```

**Review by Bas950 — CHANGES_REQUESTED** [review]

**Comment by iiTxicz** [comment]
Everything should be good now

**Review by Bas950 — APPROVED** [review]

**Review by theusaf — APPROVED** [review]

**Pull request merged**